File: app/src/main/java/com/sjn/taggingplayer/constant/ShuffleState.java
package com.sjn.taggingplayer.constant;

public enum ShuffleState implements QueueState {
    SHUFFLE(0),
    NONE(1);
    final int mNo;

    ShuffleState(int no) {
        mNo = no;
    }

    public int getNo() {
        return mNo;
    }

    public ShuffleState toggle() {
        int no = (mNo + 1) % ShuffleState.values().length;
        return ShuffleState.of(no);
    }

    public static ShuffleState of(int no) {
        for (ShuffleState state : ShuffleState.values()) {
            if (state.mNo == no) return state;
        }
        return null;
    }

    public static ShuffleState getDefault() {
        return NONE;
    }
}

File: app/src/main/java/com/sjn/stamp/constant/ShuffleState.kt
package com.sjn.stamp.constant

enum class ShuffleState(val no: Int) : QueueState {
    SHUFFLE(0),
    NONE(1);

    fun toggle(): ShuffleState? {
        val no = (this.no + 1) % ShuffleState.values().size
        return ShuffleState.of(no)
    }

    companion object {

        fun of(no: Int): ShuffleState? = ShuffleState.values().firstOrNull { it.no == no }

        fun getDefault(): ShuffleState = NONE
    }
}